src/coreclr/src/jit/codegenarm.cpp

```diff
@@ -998,13 +998,38 @@ void CodeGen::genCodeForStoreLclFld(GenTreeLclFld* tree)
     // Ensure that lclVar nodes are typed correctly.
     assert(!varDsc->lvNormalizeOnStore() || targetType == genActualType(varDsc->TypeGet()));
 
-    GenTree*    data = tree->gtOp1;
-    instruction ins  = ins_Store(targetType);
-    emitAttr    attr = emitTypeSize(targetType);
+    GenTree* data = tree->gtOp1;
+    emitAttr attr = emitTypeSize(targetType);
 
     assert(!data->isContained());
     genConsumeReg(data);
-    emit->emitIns_S_R(ins, attr, data->GetRegNum(), varNum, offset);
+    regNumber dataReg = data->GetRegNum();
+    if (varTypeIsFloating(targetType) && ((offset % emitTypeSize(TYP_FLOAT)) != 0))
+    {
+        // Arm supports unaligned access only for integer types,
+        // convert the storing floating data into 2 integer registers and write them as int.
```

Comment:
This comment should be?
`// convert the storing floating data into 1 or 2 integer registers and write them as int.`